core/src/main/scala/cats/data/Prod.scala

```diff
@@ -63,6 +74,34 @@ private[data] sealed abstract class ProdInstances4 {
   }
 }
 
+private[data] sealed abstract class ProdInstances5 extends ProdInstances6 {
+  implicit def catsDataMonadForProd[F[_], G[_]](implicit FM: Monad[F], GM: Monad[G]): Monad[λ[α => Prod[F, G, α]]] = new ProdMonad[F, G] {
+    def F: Monad[F] = FM
+    def G: Monad[G] = GM
+  }
+}
+
+private[data] sealed abstract class ProdInstances6 extends ProdInstances7 {
+  implicit def catsDataFoldableForProd[F[_], G[_]](implicit FF: Foldable[F], GF: Foldable[G]): Foldable[λ[α => Prod[F, G, α]]] = new ProdFoldable[F, G] {
+    def F: Foldable[F] = FF
+    def G: Foldable[G] = GF
+  }
+}
+
+private[data] sealed abstract class ProdInstances7 extends ProdInstances8 {
+  implicit def catsDataTraverseForProd[F[_], G[_]](implicit FF: Traverse[F], GF: Traverse[G]): Traverse[λ[α => Prod[F, G, α]]] = new ProdTraverse[F, G] {
+    def F: Traverse[F] = FF
+    def G: Traverse[G] = GF
+  }
+}
+
+private[data] sealed abstract class ProdInstances8 {
+  implicit def catsDataMonadCombineForProd[F[_], G[_]](implicit FF: MonadCombine[F], GF: MonadCombine[G]): MonadCombine[λ[α => Prod[F, G, α]]] = new ProdMonadCombine[F, G] {
+    def F: MonadCombine[F] = FF
+    def G: MonadCombine[G] = GF
+  }
+}
+
```

Comment:
In this particular case, to ensure the more specific instance having the highest priority, we basically want to ensure that instances of typeclasses who are lower in the inheritance tree are also lower in the instance trait inheritance tree. For example, `Monad` inherits `Functor` and thus is more "specific" than `Functor`, so it should be in an instance trait with higher priority (0 - 3)  than `Functor`. 
So, we need to rearrange the instances to that order. 
`MonadCombine` <: `Altenative`
`MonadCombine` <: `Monad` <: `Applicative` <: `Apply` <: `Functor`
`Traverse` <: `Foldable`
`Traverse` <: `Functor`